presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/SingleMarkDistinctToGroupBy.java

```diff
@@ -134,6 +134,8 @@ public Pattern getPattern()
         return new AggregationNode.Aggregation(
                 new FunctionCall(call.getName(), call.getWindow(), false, call.getArguments()),
                 aggregation.getSignature(),
-                Optional.empty());
+                Optional.empty(),
```

Comment:
That's construct is meaningless and/or ambiguous. It could be interpreted as:
* Values of xxxx are sorted according to yyy and first. DISTINCT is applied next. Since DISTINCT is not order-sensitive, this is a no-op.
* DISTINCT is applied first on values of xxxx. But then, the ORDER BY part is non-sensical, since values of yyy are lost.

A similar problem happens at the top-level:

```
SELECT DISTINCT a
FROM t
ORDER BY b
```

This is disallowed for the same problems I described above. Specifically, it fails with "For SELECT DISTINCT, ORDER BY expressions must appear in select list".

In conclusion, we shouldn't allow it and it should fail with a similar error.